docs/scipp_widgets_intro.ipynb

```diff
@@ -0,0 +1,184 @@
+    "For demonstration purposes we create a simple set of [scipp](https://scipp.github.io/tutorials/multi-d-datasets.html) data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "d = sc.Dataset(\n",
+    "    {\n",
+    "    'alice': sc.Variable(['z', 'y', 'x'], values=np.random.rand(10, 10, 10),\n",
+    "                         variances=0.1*np.random.rand(10, 10, 10)),\n",
+    "    'bob': sc.Variable(['x', 'z'], values=np.arange(0.0, 10.0, 0.1).reshape(10, 10),\n",
+    "                       variances=0.1*np.random.rand(10, 10))\n",
+    "    },\n",
+    "    coords={\n",
+    "        'x': sc.Variable(['x'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'y': sc.Variable(['y'], values=np.arange(11.0), unit=sc.units.m),\n",
+    "        'z': sc.Variable(['z'], values=np.arange(11.0), unit=sc.units.m)})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a display function\n",
+    "\n",
+    "scipp-widgets allows functions to be wrapped in simple graphical interfaces. As a first example take the `scipp.plot.plot` function. This takes a scipp object and plots it. In its simplest form this only has one input the object to be plotted. \n",
+    "\n",
+    "We will need to create an input object for this describing what is to be input, how we want it displayed and which input of the underlying function it should corrospond to.\n",
+    "\n",
+    "The minimum we need here is the name of the underlying function input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_input = EvalInput('scipp_obj')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The `EvalInput` class is a special Input class which before any other processing steps runs it's inputs in the notebooks scope. This allows global variables or the retults of evaluating expressions in the notebooks global scope to be passed to the underlying function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A widget which wraps the plot function can then be constructed like so"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DisplayWidget(sc.plot.plot, 'Plot', [data_input])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here the inputs are: function to wrap, operation display name, list of required input objects."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Wrapping a processing function\n",
+    "Another common case it to wrap a function which processes an input and produces an output. This can be achieved using the `ProcessWidget` class. As an example we construct a wrapper around the `scipp.sum` function below."
```

Comment:
```suggestion
    "Another common case is to wrap a function which processes an input and produces an output. This can be achieved using the `ProcessWidget` class. As an example, we construct a wrapper around the `scipp.sum` function below."
```